### PR TITLE
docs: split files into subfolders

### PR DESCRIPTION
**Pull request by Exirel**
### Description

Tin.

This is only a "first step" to reorganize the documentation around better chapters, and to rationalize where we put code references (i.e. documentation from docstrings).

I didn't change any text, it's really just moving things around.

This was initially based on #2237, #2226, and #2232, until they got merged.

### Checklist

- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

**Comment by dgw** [comment]
In lieu of a full review, which I don't really have the steam/spoons/whatever for at this moment, I would like to request that you create a patch for `sopel.chat`'s `_redirects` file that will direct requests for the deleted/moved pages to the correct new locations. Not necessarily immediately, but before this is merged. :)

**Comment by Exirel** [comment]
> I would like to request that you create a patch for `sopel.chat`'s `_redirects` file

That's fair, I'll see it done.

File: docs/source/package/plugins.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
+=========================
+Plugin Internal Machinery
+=========================
+
+.. important::
+
+   This section contains modules and classes used by Sopel internally. They are
+   subject to rapid changes between versions. They are documented here for
+   completeness, and for the aid of Sopel’s core development.
```
Comment (dgw) [inline docs/source/package/plugins.rst:9]: Try this on for size?

```suggestion
   This section contains modules and classes used by Sopel internally. They are
   subject to rapid changes between versions, without advance warning. This
   section exists primarily for the aid of Sopel’s core development.
```

File: docs/source/package/tools.rst [hunk 2]
```diff
@@ -0,0 +1,15 @@
+================
+Additional Tools
+================
+
+.. toctree::
+
+   tools/tools
```
Comment (dgw) [inline docs/source/package/tools.rst:7]: I think we can plan to have the stuff in `sopel.tools` documented here, without the redundant file path?

Personally I'm not a huge fan of having too many pages that contain only a TOC of subpages.

File: docs/source/tests/pytest.rst [hunk 3]
```diff
@@ -0,0 +1,6 @@
+=====================
+Fixtures with py.test
+=====================
```
Comment (dgw) [inline docs/source/tests/pytest.rst:3]: The `py.test` name is a legacy holdover from, oh, _years_ ago, before `pytest` was a separate project and still lived inside the `py` library.

The `py.test` package on PyPI tells users they should install with `pytest` instead, and was last touched in 2014. I guess technically the current version of `pytest` still supports the old `py.test` alias, but it's become just that: an alias. The tool's name is now `pytest`.

```suggestion
====================
Fixtures with pytest
====================
```

**Review by dgw — COMMENTED** [review]
Found some spoons. Didn't actually find that much to say, though. 😁

File: docs/source/package/tools.rst (hunk 2 above)
Comment (Exirel) [reply docs/source/package/tools.rst:7]: Yeah I'm not very happy with that one but it's a lot of things put in the wrong places and everything is more complicated than it should be.

**Comment by Exirel** [comment]
> I'll see it done.

But not yet, apparently.

**Comment by Exirel** [comment]
> But not yet, apparently.

And **now** it is done! See https://github.com/sopel-irc/sopel.chat/pull/38 for more information.

File: docs/source/package/tools.rst (hunk 2 above)
Comment (Exirel) [reply docs/source/package/tools.rst]: I reverted that back so there is no tools/tools anymore.

**Comment by Exirel** [comment]
@dgw you can review now! And remember, this is only a first step. There are so much more to come in term of documentation!

**Comment by dgw** [comment]
I'm gonna trust GitHub when it says I've still "viewed" all the changed files, and therefore trust that the _diff_ is the same as what was approved. Rebasing pulled in way too many changes from the base branch for me to sort through those. 😅

(Yes, I still wish you wouldn't rebase "just because", when there's no merge conflict, @Exirel.)

**Comment by Exirel** [comment]
I didn't rebase on the last version of master, IIRC, but I have to if I want to squash commits, don't I?

**Comment by dgw** [comment]
You don't _have to_ rebase onto any particular commit, but it's likely that you'll get "the last common commit between this branch and master" by starting your rebase with <code>git rebase -i \`git merge-base HEAD master\`</code> (note backticks; this is for Bash).

At least, I tried that on one of my somewhat stale topic branches and it started a rebase against the expected commit. YMMV as with most one-liners.

**Pull request merged**